pkg/ring/kv/dynamodb/client.go

```diff
@@ -185,6 +190,12 @@ func (c *Client) CAS(ctx context.Context, key string, f func(in interface{}) (ou
 			return c.kv.Batch(ctx, putRequests, deleteRequests)
 		}
 
+		if len(putRequests) == 0 && len(deleteRequests) == 0 {
+			// no change detected, retry
+			level.Error(c.logger).Log("msg", "finding difference", "key", key, "err", errNoChangeDeleted)
```

Comment:
Is this considered as an error? Or it is a common case.
I am thinking about removing the log. If it is common case then the log might be too verbose

Reply:
It is not a common case, it only happens when we switch states too fast for the instance. In the find difference function, this switch will be rejected, because the instance's timestamp haven't changed. https://github.com/cortexproject/cortex/blob/master/pkg/ring/model.go#L751

So we keep retry, until find difference detect the change in f function. This is similar to what memberlist does. https://github.com/cortexproject/cortex/blob/master/pkg/ring/kv/memberlist/memberlist_client.go#L838

Reply:
> It is not a common case, it only happens when we switch states too fast for the instance

Umm, why switching state too fast is not common? I feel every instance might hit this.
Anyway, when it happens, is it considered an error? From what you said it is an edge case that might happen so it is not an error to me.

I don't have issue adding this log. But better to use `warning` or even `debug` level. Maybe it is better to log something like `failed to find difference, retry cas`?

Reply:
For the instance with low or no load, they will hit this. But after 1 sec of wait, this situation should be resolved. How about changing it to info level?

Reply:
If this will be triggered when the load is low, I think it is very common, right?
I prefer to not log it at all. We already handled it with the retry

Reply:
discussed with @yeya24 offline, will remove this log.